src/Algolia.Search.Test/RetryStrategyTest/RetryStrategyTest.cs

```diff
@@ -38,6 +39,48 @@ namespace Algolia.Search.Test.RetryStrategyTest
     [Parallelizable]
     public class RetryStrategyTest
     {
+        [Test]
+        [Parallelizable]
+        public async Task TestRetryStrategyEndToEnd()
+        {
+            // Create a index with a valid client
+            var indexName = TestHelper.GetTestIndexName("test_retry_e2e");
+            var index = BaseTest.SearchClient.InitIndex(indexName);
+            var res = await index.SaveObjectAsync(new { title = "title" }, autoGenerateObjectId: true);
+            res.Wait();
+
+            // Create a client with a bad host to test that the retry worked as expected
+            var hosts = new List<StatefulHost>
+            {
+                // Bad host, will fail with
+                // System.Net.Http.HttpRequestException: nodename nor servname provided, or not known
+                new StatefulHost
+                {
+                    Url = $"{TestHelper.ApplicationId1}-1.algolianet.co",
```

Comment:
is that `.co` or `.com`?

Reply:
It's for the test, I am putting a wrong domain to have the retryStrategy to trigger and make sure the stream isn't disposed between the two calls :)

Reply:
Got it, any chance we can rather use one triggering an invalid certificate? Like `expired.badssl.com` ?

Reply:
Here we go 105024a !